ipatests/test_integration/test_ipahealthcheck.py

```diff
@@ -0,0 +1,507 @@
+            ],
+            raiseonerr=False,
+        )
+        contents = self.master.get_file_contents(
+            HEALTHCHECK_OUTPUT_FILE, encoding="utf-8"
+        )
+        data = json.loads(contents)
+        for check in data:
+            assert check["result"] == "SUCCESS"
+            assert check["kw"]["configfile"] == TOMCAT_CFG
+            assert check["kw"]["key"] in DEFAULT_PKI_CA_CERTS
+        self.master.run_command(["mv", TOMCAT_CFG, TOMCAT_CFG + ".old"])
+        result = self.master.run_command(
+            [
+                "ipa-healthcheck",
+                "--source",
+                "ipahealthcheck.dogtag.ca",
+                "--check",
+                "DogtagCertsConfigCheck",
+                "--output-file",
+                HEALTHCHECK_OUTPUT_FILE,
+            ],
+            raiseonerr=False,
+        )
+        assert result.returncode == 1
+        contents = self.master.get_file_contents(
+            HEALTHCHECK_OUTPUT_FILE, encoding="utf-8"
+        )
+        data = json.loads(contents)
+        assert data[0]["result"] == "CRITICAL"
+        self.master.run_command(["mv", TOMCAT_CFG + ".old", TOMCAT_CFG])
+        self.master.run_command(["ipactl", "restart"])
+
+    def test_source_ipahealthcheck_meta_core_metacheck(self):
+        """
+        Testcase checks behaviour of check MetaCheck in source
+        ipahealthcheck.meta.core when run on IPA master
+        """
+        self.master.run_command(
+            [
+                "ipa-healthcheck",
+                "--source",
+                "ipahealthcheck.meta.core",
+                "--check",
+                "MetaCheck",
+                "--output-file",
+                HEALTHCHECK_OUTPUT_FILE,
+            ],
+            raiseonerr=False,
+        )
+        contents = self.master.get_file_contents(
+            HEALTHCHECK_OUTPUT_FILE, encoding="utf-8"
+        )
+        data = json.loads(contents)
+        assert data[0]["result"] == "SUCCESS"
+        if osinfo.id == "fedora" and osinfo.version_number == (30):
+            assert data[0]["kw"]["ipa_version"] == "4.8.3"
+            assert data[0]["kw"]["ipa_api_version"] == "2.235"
+        else:
+            assert data[0]["kw"]["ipa_version"] == "4.9"
+            assert data[0]["kw"]["ipa_api_version"] == "2.235"
+
+    def test_source_ipahealthcheck_ipa_host_check_ipahostkeytab(self):
+        """
+        Testcase checks behaviour of check IPAHostKeytab in source
+        ipahealthcheck.ipa.host when dirsrv service is stopped and
+        running on IPA master
+        """
+        msg = (
+            "Failed to obtain host TGT: Major (851968): "
+            "Unspecified GSS failure.  "
+            "Minor code may provide more information, "
+            "Minor (2529638972): Generic error (see e-text)"
+        )
+        dashed_domain = self.master.domain.realm.replace(".", "-")
+        dirsrv_service = "dirsrv@%s.service" % dashed_domain
```

Comment:
Instead of dashed_domain:

from ipapython.ipaldap import realm_to_serverid
serverid = realm_to_serverid(api.env.realm)
dirsrv_service = "dirsrv@%s.service" % serverid